vllm/engine/arg_utils.py

```diff
@@ -501,8 +516,11 @@ def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
         guided_decoding_group.add_argument(
             "--reasoning-parser",
             # This choices is a special case because it's not static
+            # By default, no option is selected
             choices=list(ReasoningParserManager.reasoning_parsers),
-            **guided_decoding_kwargs["reasoning_backend"])
+            **(guided_decoding_kwargs["reasoning_backend"] | {
+                "default": ""
+            }))
```

Comment:
Is this change necessary? `default=None` is already in `guided_decoding_kwargs["reasoning_backend"]`